classes/various/Quaternion.sc

```diff
@@ -107,6 +107,39 @@ Quaternion {
 	printOn { arg stream;
 		stream << "Quaternion(" << a << ", " << b << ", " << c << ", " << d << ")";
 	}
+
+	// conversion to euler angles
+	// Math taken from https://en.wikipedia.org/wiki/Conversion_between_quaternions_and_Euler_angles#Quaternion_to_Euler_Angles_Conversion
+
+	// roll
+	tilt {
+		^atan2((2 * (a * b + (c * d))), (1 - (2*(b.squared + c.squared))))
+	}
+
+	// pitch
+	tumble {
+		var x, ang, val;
+		val = (2 * (a * c - (b * d))).clip(-1.0, 1.0);
+		ang = asin(val);
+		// select which axis is our reference for tumble (used to determine the quadrant)
+		x = ((this.rotate < 0.5pi) and: (this.rotate > -0.5pi)).if({
```

Comment:
it is much more efficient to enclose the part after the `and:` in curly brackets.